### PR TITLE
pallet-evm: default implementation for FeeCalculator and ConvertAccountId and separate gas price

### PR DESCRIPTION
**Pull request by sorpaas**
* Provide default implementation for `FeeCalculator` and `ConvertAccountId`, so that using the EVM module is easier.
* Move gas price parameter to dispatch parameter, and change `FeeCalculator` to output minimal gas price. This allows `FeeCalculator` to depend on storage values and thus more flexible.

### CODE REVIEW

File: frame/evm/src/lib.rs [hunk 1]
```diff
@@ -194,13 +217,14 @@ decl_module! {
 			Ok(())
 		}
 
-		#[weight = WeightForCallCreate::<T::FeeCalculator>::default()]
-		fn call(origin, target: H160, input: Vec<u8>, value: U256, gas_limit: u32)
-			-> dispatch::Result
-		{
+		#[weight = WeightForCallCreate]
+		fn call(
+			origin, target: H160, input: Vec<u8>, value: U256, gas_limit: u32, gas_price: U256,
```
Comment (gavofyork) [inline frame/evm/src/lib.rs]: style nit: it's `(long, long, long)` or `(
    high,
    high,
    high
)`. not a combination.

File: frame/evm/src/lib.rs [hunk 2]
```diff
@@ -247,11 +271,14 @@ decl_module! {
 			ret
 		}
 
-		#[weight = WeightForCallCreate::<T::FeeCalculator>::default()]
-		fn create(origin, init: Vec<u8>, value: U256, gas_limit: u32) -> dispatch::Result {
+		#[weight = WeightForCallCreate]
+		fn create(
+			origin, init: Vec<u8>, value: U256, gas_limit: u32, gas_price: U256,
```
Comment (gavofyork) [inline frame/evm/src/lib.rs]: same

File: frame/evm/src/lib.rs [hunk 3]
```diff
@@ -59,6 +63,25 @@ pub trait ConvertAccountId<A> {
 	fn convert_account_id(account_id: &A) -> H160;
 }
 
+/// Truncate the account id, taking the last 160-bit as the Ethereum address.
+pub struct TruncateConvertAccountId;
+
+impl<A: AsRef<[u8]>> ConvertAccountId<A> for TruncateConvertAccountId {
```
Comment (gavofyork) [inline frame/evm/src/lib.rs]: this might not be particularly secure without a hash first.

Comment (sorpaas) [reply frame/evm/src/lib.rs]: I changed this to `HashTruncateConvertAccountId` -- it takes a `Hasher`, hash the account id first, and then do the truncate operation.

**Pull request merged**